src/smtencoding/FStar.SMTEncoding.Term.fs

```diff
@@ -220,7 +220,41 @@ type decl =
   | SetOption  of string * string
   | GetStatistics
   | GetReasonUnknown
-type decls_t = list<decl>
+
+type decls_elt = {
+  sym_name:   option<string>;
+  key:        option<string>;
+  decls:      list<decl>;
+  a_names:    list<string>;
+}
```

Comment:
An explanation of this type would be useful

Reply:
Added, https://github.com/FStarLang/FStar/commit/38f915c2fbc808fe40a2561fa65ea4b0084c393e#diff-2907f4b54fb9d26e875c26cab34596ba.